streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -641,6 +678,19 @@ public Serde valueSerde() {
         return parsed;
     }
 
+    @Override
+    public Long getLong(final String key) {
+        // return different default value for `commit.interval.ms` if EOS enabled
+        if (StreamsConfig.COMMIT_INTERVAL_MS_CONFIG.equals(key)
+            && eosEnabled
+            // use EOS default value only if no user config is set
+            && !originals().containsKey(StreamsConfig.COMMIT_INTERVAL_MS_CONFIG)) {
+            return EOS_DEFAULT_COMMIT_INTERVAL_MS;
```

Comment:
I can change `AbstractConfig#values` from `private` to `protected` and mutate it in the constructor. However `values` would not be unmutable anymore (that is why I did not do this originally). Doing the overwrite where the value is use seems to be error prone -- it should be done in a single place and not each time the value is received from the config... A user of StreamsConfig should not need to think about this.

I also think, logging is not required, as it's just two different default values -- we never overwrite a user specified value (and it's documented in `COMMIT_INTERVAL_MS_DOC` that we do have two different default values.